src/main/java/com/checkmarx/flow/service/SastScanner.java

```diff
@@ -146,7 +146,12 @@ public ScanResults scanCli(ScanRequest request, String scanType, File... files)
         }
         return scanResults;
     }
-    
+
+    @Override
+    public ScanResults getLatestScanResults(ScanRequest request) {
+        return getLatestScanResultsAsync(request, null).join();
```

Comment:
I don't understand, join will wait for the other thread, so:
- if we are in an un-sync, I would expect not to wait (here)
- if we are in a sync, I would expect to get (not join)

Reply:
getLatestScanResultsAsync is used in webhook flows. getLatestScanResults is for command line flow.
I see no difference between get and join here, except that for 'get' we'll need to handle exceptions explicitly (which we don't care much about here).